File: VB/Dashboard_ConfigureDataConnection/Form1.vb
Imports System.Windows.Forms
Imports DevExpress.DataAccess.ConnectionParameters

Namespace Dashboard_ConfigureDataConnection

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            ' Loads a dashboard from an XML file.
            dashboardViewer1.LoadDashboard("..\..\Data\nwindDashboard.xml")
        End Sub

        ' Handles the ConfigureDataConnection event.
        Private Sub dashboardViewer1_ConfigureDataConnection(ByVal sender As Object, ByVal e As DevExpress.DataAccess.ConfigureDataConnectionEventArgs)
            ' Checks the name of the connection for which the event has been raised.
            If Equals(e.ConnectionName, "nwindConnection") Then
                ' Gets the connection parameters used to establish a connection to the database.
                Dim parameters As Access97ConnectionParameters = CType(e.ConnectionParameters, Access97ConnectionParameters)
                ' Specifies the user name used to access the database file. 
                parameters.UserName = "Admin"
                ' Specifies the password used to access the database file.
                parameters.Password = "password"
            End If
        End Sub
    End Class
End Namespace

File: CS/Dashboard_ConfigureDataConnection/Form1.cs
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DashboardCommon;
using DevExpress.XtraEditors;

namespace Dashboard_ConfigureDataConnection {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            dashboardViewer1.ConfigureDataConnection += dashboardViewer1_ConfigureDataConnection;
            dashboardViewer1.LoadDashboard(@"..\..\Data\nwindDashboard.xml");
        }

        private void dashboardViewer1_ConfigureDataConnection(object sender,
            DashboardConfigureDataConnectionEventArgs e) {
            if (e.DataSourceName == "SQL Data Source 1") {
                Access97ConnectionParameters parameters =
                    (Access97ConnectionParameters)e.ConnectionParameters;
                parameters.UserName = "Admin";
                parameters.Password = "password";
            }
        }
    }
}
